src/lib.rs

```diff
@@ -41,1346 +53,988 @@ use std::{
 /// should equal `p[label].evaluate(query)`.
 pub type Evaluations<'a, F> = BTreeMap<(String, F), F>;
 
-/// A proof of satisfaction of linear combinations.
-#[derive(Clone, CanonicalSerialize, CanonicalDeserialize)]
-pub struct BatchLCProof<F: Field, PC: PolynomialCommitment<F>> {
-    /// Evaluation proof.
-    pub proof: PC::BatchProof,
-    /// Evaluations required to verify the proof.
-    pub evals: Option<Vec<F>>,
-}
-
-impl<F: Field, PC: PolynomialCommitment<F>> SemanticallyValid for BatchLCProof<F, PC> {
-    fn is_valid(&self) -> bool {
-        self.proof.is_valid()
-            && if self.evals.is_some() {
-                self.evals.as_ref().unwrap().is_valid()
-            } else {
-                true
-            }
-    }
-}
-
-/// Describes the interface for a polynomial commitment scheme that allows
-/// a sender to commit to multiple polynomials and later provide a succinct proof
-/// of evaluation for the corresponding commitments at a query set `Q`, while
-/// enforcing per-polynomial degree bounds.
-pub trait PolynomialCommitment<F: Field>: Sized {
-    /// The universal parameters for the commitment scheme. These are "trimmed"
-    /// down to `Self::CommitterKey` and `Self::VerifierKey` by `Self::trim`.
-    type UniversalParams: PCUniversalParams;
+/// Describes the interface for a homomorphic polynomial commitment scheme with values
+/// in a commitment group `G`.  
+///
+/// The simple single-polynomial evaluation proof depends on the conrete scheme, whereas
+/// more expressive opening proofs are reduced to a simple one by leveraging the homomorphic
+/// properties.
+/// The interface comes with the additional feature of splitting the verifier into a
+/// `succinct` part and `non-succinct` part.   
+pub trait PolynomialCommitment<G: Curve>: Sized {
+    /// The polynomial commitment parameters for the commitment scheme.
+    type Parameters: PCParameters<
+        G,
+        CommitterKey = Self::CommitterKey,
+        VerifierKey = Self::VerifierKey,
+        Error = Self::Error,
+    >;
     /// The committer key for the scheme; used to commit to a polynomial and then
     /// open the commitment to produce an evaluation proof.
     type CommitterKey: PCCommitterKey;
     /// The verifier key for the scheme; used to check an evaluation proof.
     type VerifierKey: PCVerifierKey;
-    /// The prepared verifier key for the scheme; used to check an evaluation proof.
-    type PreparedVerifierKey: PCPreparedVerifierKey<Self::VerifierKey> + Clone;
+    /// The verifier state; generated by succinct verify.
+    type VerifierState: PCVerifierState;
     /// The commitment to a polynomial.
-    type Commitment: PCCommitment + Default + Debug + Eq + PartialEq;
-    /// The prepared commitment to a polynomial.
-    type PreparedCommitment: PCPreparedCommitment<Self::Commitment>;
+    type Commitment: Group<ScalarField = G::ScalarField>;
```

Comment:
For the same reason as above, we will need to compute, when batching commitments, endo mul of commitments with 128 bits scalar challenges. For this reason, would be nice to define Commitment as EndoCurve too.